orderBook.js

```diff
@@ -1,5 +1,34 @@
 function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+  // If list is empty just add the order and return
+
+  let updatedBook = []
+
+  if (existingBook.length === 0) {
+    existingBook.push(incomingOrder)
+    return existingBook
+  }
+
+  if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity === incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    return updatedBook
+  }
+
+  else if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity > incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    const updatedQuantity = existingBook[0].quantity - incomingOrder.quantity
+    existingBook[0].quantity = updatedQuantity
+    return existingBook
+  }
+
+  else if ((existingBook[0].type === 'buy' && incomingOrder.type === 'sell') && (existingBook[0].quantity < incomingOrder.quantity) && (existingBook[0].price === incomingOrder.price)) {
+    let partialOrder = []
+    const remainderQuantity = incomingOrder.quantity - existingBook[0].quantity
+    incomingOrder.quantity = remainderQuantity
+    partialOrder.push(incomingOrder)
+    return partialOrder
+  }
```

Comment:
These are only checking the first order in the book. What happens when the book has many orders and the first one didn't match but the third one would?